okapi-core/src/main/java/org/folio/okapi/util/TokenCache.java

```diff
@@ -0,0 +1,104 @@
+package org.folio.okapi.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.logging.log4j.Logger;
+import org.folio.okapi.common.OkapiLogger;
+
+public class TokenCache {
+
+  public static final long DEFAULT_TTL = 3 * 60 * 1000L;
+  
+  private final Map<String, CacheEntry> cache = new HashMap<>();
+  private final long ttl;
+  
+  private static final Logger logger = OkapiLogger.get(TokenCache.class);
+  
+  public TokenCache() {
+    this(DEFAULT_TTL);
+  }
+  
+  public TokenCache(long ttl) {
```

Comment:
Javadoc should explain what unit (minute, second, millisecond?) ttl has.

Reply:
of course - will work this into https://issues.folio.org/browse/OKAPI-890